test/VoucherHubUpgrade.test.ts

```diff
@@ -0,0 +1,72 @@
+// SPDX-FileCopyrightText: 2025 IEXEC BLOCKCHAIN TECH <contact@iex.ec>
+// SPDX-License-Identifier: Apache-2.0
+
+import { loadFixture } from '@nomicfoundation/hardhat-toolbox/network-helpers';
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+import { env } from '../config/env';
+import { getDeploymentConfig } from '../deploy/deploy';
+import { mineBlockIfOnLocalFork } from '../scripts/utils/mineBlockIfOnLocalFork';
+import * as voucherHubUtils from '../scripts/voucherHubUtils';
+import { VoucherHub__factory } from '../typechain-types';
+
+describe('VoucherHub upgrade (vNEXT)', function () {
+    before(function () {
+        if (!env.IS_LOCAL_FORK) {
+            this.skip();
+        }
+    });
+
+    async function deployFixture() {
+        const voucherHubERC1967ProxyAddress = await ethers.provider
+            .getNetwork()
+            .then((network) => network.chainId.toString())
+            .then((chainId) => {
+                if (chainId != '134') {
+                    console.error('Bellecour fork network required');
+                    process.exit(1);
+                }
+                return getDeploymentConfig(Number(chainId));
+            })
+            .then((config) => config.voucherHubAddress);
+        await mineBlockIfOnLocalFork();
+        const [admin, upgrader, manager, minter] = await ethers.getSigners();
+        const voucherHub = VoucherHub__factory.connect(voucherHubERC1967ProxyAddress!, admin);
+        const previousAdmin = await ethers.getImpersonatedSigner(
+            await voucherHub.defaultAdmin(), //'0xA0C07ad0257522211c6359EC8A4EB5d21A4A1A14',
+        );
+        console.log(
+            `Transferring VoucherHub:${voucherHubERC1967ProxyAddress} ` +
+                `roles to default hardhat accounts on this forked network..`,
+        );
+        await voucherHub
+            .connect(previousAdmin)
+            .beginDefaultAdminTransfer(admin)
+            .then((tx) => tx.wait());
+        await voucherHub.acceptDefaultAdminTransfer().then((tx) => tx.wait());
+        [
+            [await voucherHub.UPGRADER_ROLE(), upgrader],
+            [await voucherHub.MANAGER_ROLE(), manager],
+            [await voucherHub.MINTER_ROLE(), minter],
+        ].forEach((x) => voucherHub.grantRole(x[0].toString(), x[1]).then((tx) => tx.wait()));
```

Comment:
```suggestion
const roles = [
    [await voucherHub.UPGRADER_ROLE(), upgrader],
    [await voucherHub.MANAGER_ROLE(), manager],
    [await voucherHub.MINTER_ROLE(), minter],
];

for (const [role, account] of roles) {
    const tx = await voucherHub.grantRole(role.toString(), account);
    await tx.wait();
}
```
A other proposition 🤷🏻‍♂️. May be more readable

Reply:
Is this one okai or do you prefer your version https://github.com/iExecBlockchainComputing/iexec-voucher-contracts/pull/58/commits/b47c58a6c77ebe4502ec19e8244e5bb34314a9a7 ?
```
        [
            { id: await voucherHub.UPGRADER_ROLE(), account: upgrader },
            { id: await voucherHub.MANAGER_ROLE(), account: manager },
            { id: await voucherHub.MINTER_ROLE(), account: minter },
        ].forEach((role) =>
            voucherHub.grantRole(role.id.toString(), role.account).then((tx) => tx.wait()),
        );
```

Reply:
good for me also